server/Arcadia.Assistant.Web/Controllers/FeedsController.cs

```diff
@@ -33,12 +34,20 @@ public FeedsController(ActorPathsBuilder pathsBuilder, IActorRefFactory actorFac
         [Route("messages")]
         [HttpGet]
         [ProducesResponseType(typeof(Message[]), StatusCodes.Status200OK)]
-        public async Task<IActionResult> GetAllMessages(CancellationToken token)
+        public async Task<IActionResult> GetAllMessages([FromQuery] DateTime? from, [FromQuery] DateTime? to, CancellationToken token)
         {
             var timeout = this.timeoutSettings.Timeout;
 
             var feedsActor = this.actorFactory.ActorSelection(this.pathsBuilder.Get("shared-feeds"));
-            var sharedFeeds = await feedsActor.Ask<SharedFeedsActor.GetFeeds.Response>(SharedFeedsActor.GetFeeds.Instance, timeout, token);
+            SharedFeedsActor.GetFeeds.Response sharedFeeds = null;
+
+            var query = new FeedsQuery();
+            if (from.HasValue && to.HasValue) {
```

Comment:
why not have conditional query and everything else (call and awaits) unconditional?